### PR TITLE
Try to support Vite application and shared shadcn/ui components.

### PR DESCRIPTION
**Pull request by evgenius1424**
I've added vite and copied some boilerplate. But it fails with

```
../../packages/ui/src/components/ui/button.tsx:5:20 - error TS2307: Cannot find module '@ui/lib/utils' or its corresponding type declarations.

5 import { cn } from "@ui/lib/utils";
```

If you have any ideas on how to fix it, please add comments to the review.

### CODE REVIEW

File: apps/vite/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+import reactLogo from "./assets/react.svg";
+import viteLogo from "/vite.svg";
+import "./App.css";
```
Comment (dan5py) [inline apps/vite/src/App.tsx]: Delete the css import

File: apps/vite/src/index.css
Comment (dan5py) [inline apps/vite/src/index.css:1]: Delete this file

File: apps/vite/src/main.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import ReactDOM from 'react-dom/client'
+import App from './App.tsx'
+import './index.css'
```
Comment (dan5py) [inline apps/vite/src/main.tsx]: Change the css import to:
```tsx
import '@repo/ui/globals.css'
```

File: apps/vite/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from 'vite'
```
Comment (dan5py) [inline apps/vite/vite.config.ts]: Add path alias:

```ts
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import path from 'node:path'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      "@ui": path.resolve(__dirname, "../../packages/ui/src")
    }
  }
})
```

File: apps/vite/postcss.config.js
Comment (dan5py) [inline apps/vite/postcss.config.js:1]: You may also need to change the file extension to `.cjs`

File: apps/vite/src/App.css
Comment (dan5py) [inline apps/vite/src/App.css:1]: Remove this file

**Review by dan5py — COMMENTED** [review]
Hi @evgenius1424, try applying the changes and it should work. Let me know 💪

File: apps/vite/vite.config.ts (hunk 3 above)
Comment (evgenius1424) [reply apps/vite/vite.config.ts]: Thanks, if was really helpful.

File: apps/vite/postcss.config.js
Comment (evgenius1424) [reply apps/vite/postcss.config.js:1]: Yep, it works.

File: apps/vite/src/App.css
Comment (evgenius1424) [reply apps/vite/src/App.css:1]: I've removed all vite defaults.

**Comment by evgenius1424** [comment]
Hello @dan5py, I really appreciate your help with Vite support and shared components. I wanted to start some monorepo but the shadcn-ui components only worked as a local setup.

Thanks for unblocking me and possibly others.

It works great.

image

**Comment by evgenius1424** [comment]
Hi @dan5py, are you planning to merge it? I think it would be a really great update for this repo to show that even different projects can share shadcn/ui components.

**Comment by starkfire** [comment]
+1 to this PR by @evgenius1424. I was trying to implement this template on a Turborepo project with a React.js + Vite app and I used this PR as a reference for solving the runtime issues I had when importing `@ui/lib/utils` (mentioned in #20).

I would just like to add though that the `tsconfig.json` within any Vite app should contain a path that points to the `ui` package itself.

```json
"compilerOptions": {
  "paths": {
    "@ui/*": ["../../packages/ui/src/*"]
  }
}
```

This will also solve issues where the editor may complain about not being able to find the modules being imported.

**Comment by magoz** [comment]
Thank you both @evgenius1424  and @starkfire . Your combined solution works like a charm!

**Comment by dan5py** [comment]
> Hi @dan5py, are you planning to merge it? I think it would be a really great update for this repo to show that even different projects can share shadcn/ui components.

@evgenius1424, actually I like the idea of having a clean repo with a single app as boilerplate. But yeah, this would help people with a different setup. I could create a simple docs website for it. I'll see what I can do.

**Comment by evgenius1424** [comment]
Okay, I think there are projects that use turborepo with vite only. Or do some sort of migration.

If we support two major players (VIte and Next) and point out that sharing works perfectly between them, it makes a lot of sense.

rm -rf apps/vite is all you need to continue working with next only.

But yeah, it's up to you.

**Pull request closed without merging**